components/Tabs/src/MDCTabBarViewController.m

```diff
@@ -178,36 +190,47 @@ - (void)setViewControllers:(NSArray<UIViewController *> *)viewControllers {
         [viewController removeFromParentViewController];
       }
     }
-    // Update the property.
-    _viewControllers = [viewControllers copy];
     // Show the newly-visible view controller.
     [self updateTabBarItems];
   }
 }
 
-- (void)setSelectedViewController:(nullable UIViewController *)selectedViewController {
-  if (_selectedViewController != selectedViewController) {
-    if (selectedViewController) {
-      NSAssert([_viewControllers containsObject:selectedViewController], @"not one of us.");
-    }
-    BOOL animated = NO;
-    UIView *oldView = _selectedViewController.view;
-    _selectedViewController = selectedViewController;
```

Comment:
Can we not just move the add childViewController from updateTabBarItems to here?

Reply:
Discussed Offline, This is required to match Apple's behavior.